vimiv/imutils/metadata.py

```diff
@@ -0,0 +1,284 @@
+    """
+
+    @abstractmethod
+    def __init__(self, _path: str) -> None:
+        """Initialize metadata handler for a specific image.
+
+        Args:
+            _path: Path to current image.
+        """
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """Get the name of the used backend.
+
+        If no backend is used, return the name of the plugin.
+        """
+
+    @property
+    @abstractmethod
+    def version(self) -> str:
+        """Get the version of the used backend.
+
+        If no backend is used, return an empty string.
+        """
+
+    @abstractmethod
+    def get_metadata(self, _keys: Sequence[str]) -> MetadataDictT:
+        """Get value of all desired keys for the current image.
+
+        If no value is found for a certain key, do not include the key in the output.
+
+        Args:
+            _keys: Keys of metadata to query the image for.
+
+        Returns:
+            Dictionary with retrieved metadata.
+        """
+
+    @abstractmethod
+    def get_keys(self) -> Iterable[str]:
+        """Get the keys for all metadata values available for the current image."""
+
+    def copy_metadata(self, _dest: str, _reset_orientation: bool = True) -> bool:
+        """Copy metadata from the current image to dest image.
+
+        Args:
+            _dest: Path to write the metadata to.
+            _reset_orientation: If true, reset the exif orientation tag to normal.
+
+        Returns:
+            Flag indicating if copy was successful.
+        """
+        raise NotImplementedError
+
+    def get_date_time(self) -> str:
+        """Get creation date and time of the current image as formatted string."""
+        raise NotImplementedError
+
+
+# Stores all registered metadata implementations.
+_registry: List[Type[MetadataPlugin]] = []
+
+
+def has_metadata_support() -> bool:
+    """Indicate if `MetadataHandler` has `get_metadata()` and `get_keys()` capabilities.
+
+    Returns:
+        True iff at least one metadata plugins has been registered.
+    """
+    return bool(_registry)
+
+
+class MetadataHandler:
+    """Handle metadata related functionalities of images.
+
+    Attributes:
+        _path: Path to current image.
+    """
+
+    def __init__(self, path: str):
+        self._path = path
+
+    @property
+    def has_copy_metadata(self) -> bool:
+        """True iff `MetadataHandler` has an implementation for `copy_metadata`."""
+        return any(
+            list(
+                map(
+                    lambda e: e.copy_metadata != MetadataPlugin.copy_metadata, _registry
+                )
+            )
+        )
+
+    @property
+    def has_get_date_time(self) -> bool:
+        """True iff `MetadataHandler` has an implementation for `get_date_time`."""
+        return any(
+            list(
+                map(
+                    lambda e: e.get_date_time != MetadataPlugin.get_date_time, _registry
+                )
+            )
+        )
```

Comment:
Same as previous, much easier with a generator expression in the any call.

Reply:
Fixed in 65c09598a828d705363e21cecbb562161b92041b